spec/rspec/core/example_group_spec.rb

```diff
@@ -140,6 +140,12 @@ def metadata_hash(*args)
         expect(child).to have_class_const("SomeParentGroup::Hash")
       end
 
+      it 'removes it\'s constants' do
+        groups = RSpec::ExampleGroups
+        groups.remove_all_constants
```

Comment:
> I implemented this test because I didn't know how to verify if an object had been garbage collected.

We actually don't care if it's garbage collected.  The point of this change isn't to trigger GC (generally you don't want to mess with GC) but to remove the class constant references to the example groups so that GC is free to collect these example groups.  We just care that the previously defined constants have been removed.

> How would this test change if I was to put it in the test's you have identified?

Something like this would work, I think:

``` ruby
it 'removes the previously assigned example group constants' do
  RSpec.describe "SomeClass" # to ensure there's at least one constant

  expect {
    world.reset
  }.to change(RSpec::ExampleGroups, :constants).to([])
end
```